subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/SpecificInstallationToolchainSpec.java

```diff
@@ -55,7 +55,7 @@ public boolean equals(Object o) {
         if (this == o) {
             return true;
         }
-        if (o == null || getClass() != o.getClass()) {
+        if (getClass() != o.getClass()) {
```

Comment:
Why did you remove the null check?

Reply:
I've made this change due to trusting IntelliJ without thinking:
<img width="696" alt="image" src="https://user-images.githubusercontent.com/6204813/187690169-2b7cb0df-c1e4-41bf-ae73-c07678df2da8.png">
Apparently it's having a bad day/version ... (It's suggesting that the null check is not necessary even when you make it generate the equals method.)
Reverting it.